conn.go

```diff
@@ -17,6 +17,8 @@ import (
 var (
 	errInvalidWriteTopic     = errors.New("writes must NOT set Topic on kafka.Message")
 	errInvalidWritePartition = errors.New("writes must NOT set Partition on kafka.Message")
+
+	undefinedGenerationId int32 = -1
```

Comment:
It should be `const`, not `var`.

Reply:
thanks for the feedback, I applied the suggestion, I would also appreciate feedback on the general approach as well :pray: